mule-deployer/src/main/java/org/mule/tools/client/cloudhub/CloudHubClient.java

```diff
@@ -9,51 +9,52 @@
+
 import java.io.File;
+import java.lang.reflect.Type;
 import java.util.*;
-import java.util.concurrent.TimeoutException;
 
 import javax.ws.rs.client.Entity;
 import javax.ws.rs.core.GenericType;
 import javax.ws.rs.core.Response;
 
-import com.google.gson.JsonArray;
-import com.google.gson.JsonElement;
-import com.google.gson.JsonObject;
-import com.google.gson.JsonParser;
+import com.google.common.base.Preconditions;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.apache.commons.lang3.StringUtils;
+import org.glassfish.jersey.media.multipart.FormDataBodyPart;
 import org.glassfish.jersey.media.multipart.FormDataMultiPart;
 import org.glassfish.jersey.media.multipart.MultiPart;
 import org.glassfish.jersey.media.multipart.file.FileDataBodyPart;
 import org.mule.tools.client.AbstractMuleClient;
+import org.mule.tools.client.cloudhub.model.Application;
+import org.mule.tools.client.cloudhub.model.DomainAvailability;
+import org.mule.tools.client.cloudhub.model.MuleVersion;
+import org.mule.tools.client.cloudhub.model.PaginatedResponse;
+import org.mule.tools.client.cloudhub.model.SupportedVersion;
 import org.mule.tools.client.exception.ClientException;
 import org.mule.tools.model.anypoint.CloudHubDeployment;
 import org.mule.tools.utils.DeployerLog;
 
 public class CloudHubClient extends AbstractMuleClient {
 
-  public static final String STARTED_STATUS = "STARTED";
-  public static final String UNDEPLOYED_STATUS = "UNDEPLOYED";
-
-  private static final String APPLICATIONS_PATH = "/cloudhub/api/applications";
-  private static final String DOMAINS_PATH = "/cloudhub/api/applications/domains/";
-  private static final String APPLICATION_UPDATE_PATH = "/cloudhub/api/v2/applications/%s";
-  private static final String APPLICATIONS_FILES_PATH = "/cloudhub/api/v2/applications/%s/files";
-  private static final String SUPPORTED_VERSIONS_PATH = "cloudhub/api/mule-versions";
-  private static final String CREATE_REQUEST_TEMPLATE = "{" +
-      "  \"domain\": \"%s\"," +
-      "  \"region\": \"%s\"," +
-      "  \"muleVersion\": \"%s\"," +
-      "  \"workers\": %d," +
-      "  \"workerType\": \"%s\"";
-  private static final String UPDATE_REQUEST_TEMPLATE = "{" +
-      "  \"region\":\"%s\"," +
-      "  \"muleVersion\": {\"version\": \"%s\"}," +
-      "  \"workers\": {" +
-      "    \"amount\": %d," +
-      "    \"type\": {" +
-      "        \"name\": \"%s\"" +
-      "    }" +
-      "  }";
+  private static final String API_VERSION = "v2";
+  private static final String BASE_PATH = "/cloudhub/api";
+
+  // TODO using version less
+  private static final String SUPPORTED_VERSIONS_PATH = BASE_PATH + "/mule-versions";
+  private static final String APPLICATION_STATUS = BASE_PATH + "/applications/%s/status";
+  private static final String APPLICATIONS_DOMAINS_PATH = BASE_PATH + "/applications" + "/domains/%s";
+
+  private static final String BASE_API_PATH = BASE_PATH + "/" + API_VERSION;
```

Comment:
It took me a while to understand the difference between BASE_API_PATH and BASE_PATH. I'd go for BASE_API_VERSION_PATH for the former and BASE_API_PATH for the latter

Reply:
DONE